cmd/tigerblood/main.go

```diff
@@ -117,22 +127,44 @@ func main() {
 		}
 	}
 	log.Printf("loaded violation map: %s", penalties)
-	var handler http.Handler = tigerblood.NewTigerbloodHandler(db, statsdClient, penalties)
+
+	return penalties
+}
+
+func main() {
+	mozlogrus.Enable("tigerblood")
+	loadConfig()
+	printConfig()
+
+	var middleware = make([]tigerblood.Middleware, 1)
+	middleware = append(middleware, tigerblood.RecordStartTime())
 
 	if viper.GetBool("HAWK") {
-		credentials := viper.GetStringMapString("CREDENTIALS")
-		if len(credentials) == 0 {
-			log.Fatal("Hawk was enabled, but no credentials were found.")
-		} else {
-			log.Printf("Hawk enabled with %d credentials.", len(credentials))
-		}
-		handler = tigerblood.NewHawkHandler(handler, credentials)
+		credentials := loadCredentials()
+		middleware = append(middleware, tigerblood.RequireHawkAuth(credentials))
 	}
 
-	http.HandleFunc("/", handler.ServeHTTP)
-	log.Printf("Listening on %s", viper.GetString("BIND_ADDR"))
-	err = http.ListenAndServe(viper.GetString("BIND_ADDR"), nil)
-	if err != nil {
-		log.Fatal(err)
+	db := loadDB()
+	middleware = append(middleware, tigerblood.AddDB(db))
+
+	if viper.IsSet("STATSD_ADDR") {
+		statsdClient := loadStatsd()
+		middleware = append(middleware, tigerblood.AddStatsdClient(statsdClient))
+	} else {
+		log.Println("statsd not found")
 	}
+
+	penalties := loadViolationPenalties()
+	middleware = append(middleware, tigerblood.AddViolations(penalties))
+
+
+	middleware = append(middleware, tigerblood.SetResponseHeaders())
```

Comment:
Maybe future addition: a flag to turn this middleware off, just for non-mozilla users and local testing.